Source/WebCore/rendering/RenderTextControlSingleLine.cpp

```diff
@@ -366,15 +366,21 @@ void RenderTextControlSingleLine::autoscroll(const IntPoint& position)
 
 int RenderTextControlSingleLine::scrollWidth() const
 {
-    if (auto innerTextElement = this->innerTextElement(); innerTextElement && innerTextElement->renderer())
-        return innerTextElement->renderer()->scrollWidth();
+    if (auto* innerTextRenderer = innerTextElement() ? innerTextElement()->renderer() : nullptr) {
+        // Adjust scrollWidth to inculde input element horizontal paddings and decoration width.
+        auto adjustment = clientWidth() - innerTextRenderer->clientWidth();
+        return innerTextElement()->renderer()->scrollWidth() + adjustment;
```

Comment:
this should read: innerTextRenderer->scrollWidth()

Reply:
Ops! My sleepy head.. :-( Fixed now.